src/main/File/FileController.java

```diff
@@ -0,0 +1,478 @@
+          req = null;
+        }
+
+        User check = userCheck(body);
+        if (req != null && req.has("targetUser") && check == null) {
+          log.info("Target User could not be found in the database");
+          response = UserMessage.USER_NOT_FOUND;
+        } else {
+          boolean orgFlag;
+          if (req != null && req.has("targetUser") && check != null) {
+            log.info("Target user found, setting parameters.");
+            username = check.getUsername();
+            organizationName = check.getOrganization();
+            privilegeLevel = check.getUserType();
+            orgFlag = organizationName.equals(ctx.sessionAttribute("orgName"));
+          } else {
+            log.info("Checking session for user.");
+            username = ctx.sessionAttribute("username");
+            organizationName = ctx.sessionAttribute("orgName");
+            privilegeLevel = ctx.sessionAttribute("privilegeLevel");
+            orgFlag = true;
+          }
+
+          if (orgFlag) {
+            if (file == null) {
+              log.info("File is null, invalid file!");
+              response = FileMessage.INVALID_FILE;
+            } else {
+              FileType fileType = FileType.createFromString(ctx.formParam("fileType"));
+              log.info("Received file type of {}", fileType.toString());
+              String title = null;
+              boolean annotated = false;
+              boolean toSign = false;
+              if (ctx.formParam("annotated") != null) {
+                annotated = Boolean.parseBoolean(ctx.formParam("annotated"));
+              }
+              if (ctx.formParam("toSign") != null) {
+                toSign = Boolean.parseBoolean(ctx.formParam("toSign"));
+              }
+              String fileId = null;
+              UploadedFile signature = null;
+              if (fileType.isPDF()) {
+                log.info("Got PDF file to upload!");
+                try {
+                  InputStream content = file.getContent();
+                  PDDocument pdfDocument = PDDocument.load(content);
+                  title = getPDFTitle(file.getFilename(), pdfDocument);
+                  content.reset();
+                  pdfDocument.close();
+                } catch (Exception exception) {
+                  ctx.result(FileMessage.INVALID_FILE.toResponseString());
+                }
+
+                if (toSign) {
+                  signature = Objects.requireNonNull(ctx.uploadedFile("signature"));
+                }
+
+                if (fileType == FileType.FORM_PDF && annotated) {
+                  fileId = Objects.requireNonNull(ctx.formParam("fileID"));
+                }
+                UploadFileService uploadService =
+                    new UploadFileService(
+                        db,
+                        fileDao,
+                        username,
+                        organizationName,
+                        privilegeLevel,
+                        fileType,
```

Comment:
To me this constructor seems a bit too long. Can't we just pass the File object here and the UploadFileService can just get the fields out of the File?